### PR TITLE
Discarding locations if they are pretty close to the last one known.

### PR DESCRIPTION
**Pull request by arcturus**
@wilsonpage r?

### CODE REVIEW

File: MagnetScannerIOS/ScannerGeolocation.swift [hunk 1]
```diff
@@ -12,14 +12,17 @@ import SwiftyJSON
 class ScannerGeolocation: NSObject, CLLocationManagerDelegate, Scanner {
   let locationManager: CLLocationManager = CLLocationManager()
   var callback: ((Dictionary<String, AnyObject>) -> Void)!
-  var initialized: Bool = false;
+  var initialized: Bool = false
+  var lastLocation: CLLocation!
+    let MIN_DISTANCE: Double = 10
```
Comment (wilsonpage) [inline MagnetScannerIOS/ScannerGeolocation.swift]: NIT indentation

File: MagnetScannerIOS/ScannerGeolocation.swift [hunk 2]
```diff
@@ -54,6 +57,11 @@ class ScannerGeolocation: NSObject, CLLocationManagerDelegate, Scanner {
       return
     }
     
+    guard (lastLocation == nil) || lastLocation!.distanceFromLocation(location) < MIN_DISTANCE else {
+        return
+    }
+    lastLocation = location
```
Comment (wilsonpage) [inline MagnetScannerIOS/ScannerGeolocation.swift:68]: NIT: Line breaks after blocks.

File: MagnetScannerIOS/ScannerGeolocation.swift [hunk 3]
```diff
@@ -54,6 +57,11 @@ class ScannerGeolocation: NSObject, CLLocationManagerDelegate, Scanner {
       return
     }
     
+    guard (lastLocation == nil) || lastLocation!.distanceFromLocation(location) < MIN_DISTANCE else {
```
Comment (wilsonpage) [inline MagnetScannerIOS/ScannerGeolocation.swift]: This line is hard to parse, can you pull the logic out into a function:

``` swift
guard isValidNextLocation(location) else { return }
lastLocation = location
```

**Review by wilsonpage — COMMENTED** [review]
r+ after nits

**Comment by arcturus** [comment]
Thx, changes applied.

Shipping.
